ompi/mca/pml/ob1/pml_ob1_cuda.c

```diff
@@ -1,222 +0,0 @@
-
-    return num_btls_used;
-}
-
-int mca_pml_ob1_cuda_need_buffers(void * rreq,
-                                  mca_btl_base_module_t* btl)
-{
-    mca_pml_ob1_recv_request_t* recvreq = (mca_pml_ob1_recv_request_t*)rreq;
-    mca_bml_base_endpoint_t* bml_endpoint = mca_bml_base_get_endpoint (recvreq->req_recv.req_base.req_proc);
-    mca_bml_base_btl_t *bml_btl = mca_bml_base_btl_array_find(&bml_endpoint->btl_send, btl);
-
-    /* A btl could be in the rdma list but not in the send list so check there also */
-    if (NULL == bml_btl) {
-        bml_btl = mca_bml_base_btl_array_find(&bml_endpoint->btl_rdma, btl);
-    }
-    /* We should always be able to find back the bml_btl based on the btl */
-    assert(NULL != bml_btl);
-
-    if ((recvreq->req_recv.req_base.req_convertor.flags & CONVERTOR_CUDA) &&
-        (bml_btl->btl_flags & MCA_BTL_FLAGS_CUDA_GET)) {
-        recvreq->req_recv.req_base.req_convertor.flags &= ~CONVERTOR_CUDA;
-        if(opal_convertor_need_buffers(&recvreq->req_recv.req_base.req_convertor) == true) {
-            recvreq->req_recv.req_base.req_convertor.flags |= CONVERTOR_CUDA;
-            return true;
-        } else {
-            recvreq->req_recv.req_base.req_convertor.flags |= CONVERTOR_CUDA;
-            return false;
-        }
-    }
-    return true;
-}
-
-/*
- * This function enables us to start using RDMA get protocol with GPU buffers.
- * We do this by adjusting the flags in the BML structure.  This is not the
- * best thing, but this may go away if CUDA IPC is supported everywhere in the
- * future. */
-void mca_pml_ob1_cuda_add_ipc_support(struct mca_btl_base_module_t* btl, int32_t flags,
-                                      ompi_proc_t* errproc, char* btlinfo)
-{
-    mca_bml_base_endpoint_t* ep;
-    int btl_verbose_stream = 0;
-    int i;
-
-    assert(NULL != errproc);
-    assert(NULL != errproc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_BML]);
-    if (NULL != btlinfo) {
-        btl_verbose_stream = *(int *)btlinfo;
-    }
-    ep = (mca_bml_base_endpoint_t*)errproc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_BML];
-
-    /* Find the corresponding bml and adjust the flag to support CUDA get */
-    for( i = 0; i < (int)ep->btl_send.arr_size; i++ ) {
-        if( ep->btl_send.bml_btls[i].btl == btl ) {
-            if (4 < opal_output_get_verbosity(btl_verbose_stream)) {
-                char *errhost = opal_get_proc_hostname(&errproc->super);
-                opal_output(0, "BTL %s: rank=%d enabling CUDA IPC "
-                            "to rank=%d on node=%s \n",
-                            btl->btl_component->btl_version.mca_component_name,
-                            OMPI_PROC_MY_NAME->vpid,
-                            ((ompi_process_name_t*)&errproc->super.proc_name)->vpid,
-                            errhost);
-                free(errhost);
-            }
-            ep->btl_send.bml_btls[i].btl_flags |= MCA_BTL_FLAGS_CUDA_GET;
-        }
-    }
-}
```

Comment:
@wckzhang After removing this code, have you tested if PML ob1 works as before with CUDA?

Reply:
This code wasn't really removed, just moved into pml_ob1_accelerator.c and renamed.